apps/docs/src/guide/wallets/wallet-transferring.md

```diff
@@ -1,14 +1,59 @@
 # Wallet Transferring
 
-This guide demonstrates how to transfer assets between accounts and contracts, and how to validate your balance prior to a transfer.
+This guide provides instructions for transferring assets between wallets and contracts using the SDK. It includes methods for initiating transfers, handling transaction requests, and ensuring sufficient balance before transfers.
 
-## Transferring Between Wallets
+## Transferring Assets Between Accounts
 
-Transferring assets between wallets is really simple within the SDK.
+The `account.transfer` method initiates a transaction request that transfers an asset from one wallet to another. This method requires three parameters:
+
+1. The receiver's wallet address.
+2. The amount of the asset to be transferred.
+3. The ID of the asset to be transferred (Optional - Defaults to the base asset id).
+
+Upon execution, this function returns a promise that resolves to a transaction response. To wait for the transaction to be processed, call `response.wait()`.
+
+### Example
+
+Here is an illustration of how to use the `account.transfer` function:
+
+<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-1{ts:line-numbers}
+
+In the previous example, we used the `transfer` method, which essentially creates a `ScriptTransactionRequest`, populates its data with the provided transfer information, and submits the transaction request to the node.
+
+However, there may be times when you need the Transaction ID before actually submitting it to the node. To achieve this, you can simply call the `createTransfer` method instead.
+
+This method also creates a `ScriptTransactionRequest` and populates it with the provided data, but instead of submitting it to the node, it returns the request.
+
+<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-2{ts:line-numbers}
+
+Once you initiate a transaction request, any changes will create a new, separate transaction with a `different` ID. Therefore, avoid modifying the transaction request if you want to keep the same ID.
+
+<<< @/../../docs-snippets/src/guide/cookbook/transferring-assets.test.ts#transferring-assets-3{ts:line-numbers}
+
+## Transferring Assets To Contracts
+
+When transferring assets to a deployed contract, we use the `transferToContract` method, which shares a similar parameter structure with the `account.transfer` method.
+
+However, instead of supplying the target wallet's address, as done in `destination.address` for the transfer method, we need to provide an instance of [Address](../types/address.md) created from the deployed contract id.
```

Comment:
This feels like poor DX? We should have consistency whether we're passing the Object or the literal str address 🤔

Reply:
cc @FuelLabs/sdk-ts for thoughts?

Reply:
@danielbate I remember we had discussions about this in the past. We then created an issue for it: https://github.com/FuelLabs/fuels-ts/issues/1584

Reply:
There's also this:
 - https://github.com/FuelLabs/fuels-ts/issues/2348